### PR TITLE
Use lambda_provided for graalvm runtime

### LINKED ISSUE #1160: Unless the user selects GraalVM feature runtime should be LAMBDA_JAVA if GraalVM it should be LAMBDA_PROVIDED
This should be a 3.5.x change

see: https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/434

### LINKED ISSUE #1206: select correct runtime for gradle and labmda
Since Micronaut Gradle plugin 3.4.0 we support now two runtimes for Lambda. If you are deploying your Lambda functions to a Java runtime use the environment `lambda_java`. If you deploy a GraalVM Native Image of your Micronaut Function to a Lambda Provided runtime use the environment `lambda_provided`. 

We should select the appropriate runtimes when the user selects `aws-lambda` or if the users selects `aws-lambda` + `graalvm`. features.

### PR DESCRIPTION
**Pull request by timyates**
See https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/434

Fixes #1160

### CODE REVIEW

**Comment by timyates** [comment]
Failing at the moment as I was too quick... 🤣

**Comment by timyates** [comment]
Waiting for the changes to hit 3.5.0-SNAPSHOT

File: starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw [hunk 1]
```diff
@@ -58,7 +58,11 @@ MavenBuild mavenBuild
     <micronaut.runtime>azure_function</micronaut.runtime>
 }
 @if (applicationType == ApplicationType.DEFAULT && features.contains("aws-lambda")) {
-    <micronaut.runtime>lambda</micronaut.runtime>
+  @if (features.contains("graalvm")) {
+    <micronaut.runtime>lambda_provided</micronaut.runtime>
+  } else {
+    <micronaut.runtime>lambda_java</micronaut.runtime>
+  }
```
Comment (alvarosanchez) [inline starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw]: This will fail with Maven, since those values don't exist in this enum: https://github.com/micronaut-projects/micronaut-maven-plugin/blob/master/src/main/java/io/micronaut/build/MicronautRuntime.java

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw]: This is blocked via: https://github.com/micronaut-projects/micronaut-maven-plugin/issues/321

File: starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/templates/buildGradle.rocker.raw [hunk 2]
```diff
@@ -251,7 +251,7 @@ protobuf {
 protobuf {
     protoc { artifact = "com.google.protobuf:protoc:@VersionInfo.getBomVersion("protobuf")" }
     plugins {
-        grpc { artifact = "io.grpc:protoc-gen-grpc-java:@VersionInfo.getBomVersion("grpc")" }
+        grpc { artifact = "io.grpc:protoc-gen-grpc-java:@VersionInfo.getBomVersion("micronaut.grpc")" }
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/templates/buildGradle.rocker.raw]: Please, move this to a different PR.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/templates/buildGradle.rocker.raw (hunk 2 above)
Comment (timyates) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/templates/buildGradle.rocker.raw]: Done https://github.com/micronaut-projects/micronaut-starter/pull/1162

File: starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw (hunk 1 above)
Comment (timyates) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw]: Also by https://github.com/micronaut-projects/micronaut-core/pull/7190 for the grpc fix

**Comment by melix** [comment]
This will require new releases of the Gradle and Maven plugins, right?

**Comment by sdelamo** [comment]
> This will require new releases of the Gradle and Maven plugins, right?

Correct

**Review by melix — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Comment by timyates** [comment]
Is this still blocked by https://github.com/micronaut-projects/micronaut-maven-plugin/issues/321 ?

**Comment by alvarosanchez** [comment]
> Is this still blocked by [micronaut-projects/micronaut-maven-plugin#321](https://github.com/micronaut-projects/micronaut-maven-plugin/issues/321) ?

Yes. Providing "invalid" values for the `micronaut.runtime` property will result in a build error

I will work on issue 321 next on my list.

**Comment by sdelamo** [comment]
> > Is this still blocked by [micronaut-projects/micronaut-maven-plugin#321](https://github.com/micronaut-projects/micronaut-maven-plugin/issues/321) ?
> 
> Yes. Providing "invalid" values for the `micronaut.runtime` property will result in a build error
> 
> I will work on issue 321 next on my list.

Please, note I changed the pr so that for maven `lambda` runtime is still used.

**Review by alvarosanchez — APPROVED** [review]
